### DISCUSSION TITLE
Unable to load bundle identifiers

### DISCUSSION THREAD

**Discussion by TheCarpetMerchant** [codemagic-ci-cd/codemagic-docs#1929] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

- Flutter

### Steps to reproduce

I am following the documentation at https://labs.codemagic.io/your-first-flutter-app-to-appstore/#4
 Gotten this error when switching code signing on :
```
Unable to load bundle identifiers
Loading bundle identifiers for KEYNAME failed. Authentication credentials are missing or invalid. - Provide a properly configured and signed bearer token, and make sure that it has not expired. Learn more about Generating Tokens for API Requests https://developer.apple.com/go/?id=api-generating-tokens
```
So apparently, my credentials aren't valid, but I don't see how I'm supposed to _make_ them valid.

### Expected results

none

### Actual results

none

**Comment by kulakarla** [comment]
Hi @TheCarpetMerchant 

You have not added the API key successfully. Follow this documentation step [here](https://docs.codemagic.io/flutter-code-signing/ios-code-signing/#step-1-creating-an-app-store-api-key-for-codemagic) for adding the Apple Developer Portal integration successfully to Codemagic. Make sure your Issuer ID and Key ID-s match with the ones in the Developer Portal. Also, for the key name, for extra safety, use the same name as it is named in the Developer Portal.

This should solve the error, if not, let me know.

**Reply by TheCarpetMerchant** [reply]
Those are the steps I followed, with the exception that I do not have a Team on Codemagic and had to access the Developer Portal Connect button in another way. But that went well, it's only the Code Signing itself that raises this error.

**Reply by kulakarla** [reply]
Yes, you can add the integration under your personal account integrations aswell. However, you must be doing something incorrectly in that step which produces the error. What access level did you give the API Key?

**Reply by TheCarpetMerchant** [reply]
Like the documentation mentions, App Manager.
I have created a device, an ios dev certificate, a provisioning profile...

**Reply by TheCarpetMerchant** [reply, marked as answer]
Okay, I went and regenerated another key, works fine with Codemagic. The problem is clearly on Apple's side. One more reason to hate them, yey!

**Reply by RedyAu** [reply]
Exact same happened to me. Key generated at first: Nah-ah. Same process again: Now it works. Great.

**Reply by gpinkham** [reply]
ok I guess I am in the same situation   three attempts with the same results..  guess I'll give up for the day.
